### DISCUSSION TITLE
Train on many inputs (land cover)

### DISCUSSION THREAD

**Discussion by luchomedrano** [PUTvision/qgis-plugin-deepness#79] [Q&A]
I'd like to replicate a USGS land cover classification for regions outside US.

I guess one should be able to "train" a model with global coverage images, as landsat, sentinel, bing, modis etc and use as "supervised" label data the US land use categories.

Once a model is "precise" enough, it should be able to reproduce the USGS categories on any given area on the globe, and the global coverage images as input.

I'm a casual user, and therefore I ask for help here. Is there a way to use the plugin on that objective? Any suggestions?

Thanks you!

**Comment by przemyslaw-aszkowski** [comment]
Hi,
Your reasoning is generally correct, but our Deepness plugin is only aimed to use an already-trained models. Therefore you would need to train your model outside of the QGIS. Here is a short summary if you would like to try it by yourself:
1) First you need to prepare the training data (e.g. satellite images from bing and corresponding coverage categories on these images). I believe you can find a few datasets online.
2) Then you need to train and validate the model. For that, we usually use the PyTorch library.  You can look for "PyTorch image segmentation tutorials" online.
3) Then export the model to onnx format. This format is supported in our Deepness plugin.

But by the way, the land cover classification model in our ZOO is trained on the terrain in Poland, so you can use it anywhere - but it will be less precise of course.

Hope that it is helpful to you!

**Reply by jon-m-l** [reply]
Hi,
I also would like to train a new model for roof classification with deepness.
Would it be possible for you to share your training-script (for land classification) with example training data so I could produce corresponding training data and adapt the script?
Thanks!

**Reply by przemyslaw-aszkowski** [reply]
Hi, 
unfortunately at the moment, we do not have ready to share training script, as it was done in a scratchpad witch we do not have anymore. We have a plan to prepare a readable training notebook (with a tutorial) within the next few weeks. But if you already managed to do the bespoken training, please feel welcome to contribute to the repository
